### PR TITLE
[#7] Implement property test for passwords

### LINKED ISSUE #7: Add property test on passwords verification
It would be good to the check this property:
```
verifyPassword pwd (mkPasswordHash pwd) === True
```

### PR DESCRIPTION
**Pull request by vrom911**
Tests are really slow now.. Not sure if it worth to be merged now. I opened the issue in hedgehog regarding the problem I faced https://github.com/hedgehogqa/haskell-hedgehog/issues/192 , but probably you've got any ideas how to improve speed.

Should resolve #7

### CODE REVIEW

**Comment by arbus** [comment]
We can speed it up by creating a higher level function that takes in a hashing policy and using the faster hashing policy just for the tests

File: test/Test/Password.hs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+module Test.Password where
+
+import Control.Monad.Except (MonadError)
+import Lib.App.Error (AppError (..))
+
+import Hedgehog (MonadGen, Property, forAll, property, withTests, (===))
+import Test.Tasty
+import Test.Tasty.Hedgehog
+
+import Lib.Util.Password
```
Comment (ChShersh) [inline test/Test/Password.hs]: Implicit imports?..

File: test/Test/Password.hs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+
+test_pwdHashVerified :: [TestTree]
+test_pwdHashVerified = pure $ testProperty "Password verification" pwdProperty
+
+pwdProperty :: Property
+pwdProperty  = property $ do
```
Comment (ChShersh) [inline test/Test/Password.hs]: Looks like extra space here to me

File: test/Test/Password.hs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  randomPwd <- forAll genPwd
+  whenRightM (runExceptT $ mkPasswordHash randomPwd) $ \pwdHash ->
+    verifyPassword randomPwd pwdHash === True
```
Comment (ChShersh) [inline test/Test/Password.hs]: I think it's better to use [`assert`](https://hackage.haskell.org/package/hedgehog-0.5.3/docs/Hedgehog.html#v:assert) here. But this is really minor.

**Comment by ChShersh** [comment]
Also, having high-level function which takes policy looks like a good temporary solution to me so that's okay 👌

**Pull request merged**